### DISCUSSION TITLE
Working with roles that aren't open source

### DISCUSSION THREAD

**Discussion by nirvdrum** [ansible/ansible-lint#3982] [Q&A]
I'm in the process of updating an Ansible playbook I haven't touched in a year and I'm now getting validation errors about my role metadata requiring a `galaxy_info.license` value. These roles are being used privately and, as such, aren't licensed. I went through the list of licenses defined in the schema and couldn't find anything that corresponds to private, non-licensed code (not to be confused with ["unlicensed" code](https://choosealicense.com/licenses/unlicense/)).

Setting the `license` field to be `null` seems to me to be a more accurate representation of my role's licensed status, but the schema disallows that. For now, I'm just setting the value to an empty string. There isn't any validation of the field contents, which is nice, but what's the recommended way of dealing with this situation?

**Comment by Tualua** [comment]
Hello! Did you find a solution?

**Reply by nirvdrum** [reply]
I never received an answer, unfortunately. I'm still using an empty string as the license value.

**Comment by jayhendren** [comment]
FWIW, the schema (and the list of SPDX licenses therein) in `src/ansiblelint/schemas/galaxy.json` only applies to `galaxy.yml`, not `meta/main.yml`.  As you pointed out, ansible-lint only cares that the `license:` field in `meta/main.yml` is a string, but not that the string has any particular value.

IANAL, but I kind of like `license: "All Rights Reserved"`, as it is more explicit in its intention.  But it is my understanding that, from a legal perspective, `license: ""` does exactly what you want (i.e., grants no license).  Perhaps `license: "None"` could work as well?
